apps/app.go

```diff
@@ -135,7 +136,7 @@ func marathonAppNameToServiceName(name string, nameSeparator string) string {
 }
 
 func labelsToTags(labels map[string]string) []string {
-	tags := []string{}
+	tags := make([]string, 0, len(labels))
```

Comment:
I'm prepare the benchmark to see the whole change in performance of `RegistrationIntents` not just `labelsToTags`.
In this PR I removed one allocation in creation of tags from labels. Instead of starting with empty slice and grow it when needed I start with capacity that will be sufficient for tags. This reduces the allocation of the caller. 

I do not agree this is ineffective but I agree it's a micro optimization. `labelsToTags` is called twice and this change improves the first call. There are cases when the second call will also be improved (port labels count equal size of common tags but none of this label will be a tag) but I'm only interested in most common case in our installation. 

If you think you can improve this part of code feel free to use the benchmark and show your results. I'll be happy to accept a pull request.